splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -367,6 +370,51 @@ public void createKeyColumnUseViewInSysIBM(TransactionController tc) throws Stan
         SpliceLogUtils.info(LOG, "View SYSKEYCOLUSE in SYSIBM is created!");
     }
 
+    private TabInfoImpl getNaturalNumbersTable() throws StandardException{
+        if(naturalNumbersTable==null){
+            naturalNumbersTable=new TabInfoImpl(new SYSNATURALNUMBERSRowFactory(uuidFactory,exFactory,dvf, this));
+        }
+        initSystemIndexVariables(naturalNumbersTable);
+        return naturalNumbersTable;
+    }
+
+    /**
+     * Populate SYSNATURALNUMBERS table with 1-2048.
+     *
+     * @throws StandardException Standard Derby error policy
+     */
+    private void populateSYSNATURALNUMBERS(TransactionController tc) throws StandardException{
+        TabInfoImpl ti = getNonCoreTI(SYSNATURALNUMBERS_CATALOG_NUM);
+        for (int i = 1; i <= 2048; i++) {
+            NaturalNumberDescriptor nnd = new NaturalNumberDescriptor(i);
+            ExecRow row = ti.getCatalogRowFactory().makeRow(nnd, null);
+            // ignore return value because sysnaturalnumbers does not have indexes
+            ti.insertRow(row, tc);
+        }
```

Comment:
shouldn't we maybe put this in SYSNATURALNUMBERSRowFactory.java? Otherwise SpliceDataDictionary one day will be 20'000 lines long, basically taking care of "everything".

Reply:
Good point! I move the main logic to the row factory. However, is wrapper has to be in `SpliceDataDictionary` because we need to get the `TabInfoImpl` from data dictionary's`nonCoreTI`.